presto-main/src/main/java/com/facebook/presto/execution/TaskExecutor.java

```diff
@@ -87,6 +88,30 @@
 
     private final Ticker ticker;
 
+    private static class SplitProfile
+        implements Comparable<SplitProfile>
+    {
+        private final long startTime;
+        private final long splitId;
+
+        private SplitProfile(long startTime, long splitId)
```

Comment:
Make the constructor public, since you are using it outside of this class. A private constructor only makes sense if it's called by a static factory method within the same class.

(The visibility rules for nested classes is strange and allows this behavior, but it's best to write the code given the logical intent, even if Java allows the outer class to bypass the visibility.)